enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/InternalFlatFileRealm.java

```diff
@@ -546,19 +529,29 @@ private void checkValidityOfUsernameAndRoleName( String username, String roleNam
 
     private void assertValidUsername( String name ) throws InvalidArgumentsException
     {
+        if ( name.isEmpty() )
+        {
+            throw new InvalidArgumentsException( "The provided user name is empty." );
+        }
         if ( !userRepository.isValidUsername( name ) )
         {
             throw new InvalidArgumentsException(
-                    "User name contains illegal characters. Please use simple ascii characters and numbers." );
+                    "User name '" + name +
+                    "' contains illegal characters. Use simple ascii characters and numbers." );
```

Comment:
Feels a bit sad, but I took it from [here](https://msdn.microsoft.com/en-us/library/ms679325(v=vs.85).aspx). It says 

> Avoid the word "please". It can be interpreted to mean that a required action is optional.